coreax/metrics.py

```diff
@@ -39,472 +39,93 @@
 _Data = TypeVar("_Data", bound=coreax.data.Data)
 
 
-class Metric(ABC, Generic[_Data]):
+class Metric(eqx.Module, Generic[_Data]):
     """Base class for calculating metrics."""
 
     @abstractmethod
-    def compute(
-        self,
-        reference_data: _Data,
-        comparison_data: _Data,
-    ) -> Array:
+    def compute(self, reference_data: _Data, comparison_data: _Data, **kwargs) -> Array:
         r"""
-        Compute the metric.
-
-        Return a zero-dimensional array.
+        Compute the metric/distance between the reference and comparison data.
 
         :param reference_data: An instance of the class :class:`coreax.data.Data`,
-            containing an :math:`n \times d` array of data. Or an instance of the class
-            :class:`coreax.data.SupervisedData` containing an :math:`n \times d` array
-            of data, and a corresponding :math:`n \times p` array of supervision.
+            containing an :math:`n \times d` array of data
         :param comparison_data: An instance of the class :class:`coreax.data.Data` to
-            compare against ``reference_data`` containing an :math:`m \times d` array of
-            data. Or an instance of the class :class:`coreax.data.SupervisedData` to
-            compare against ``reference_data`` containing an :math:`m \times d` array of
-            data, and a corresponding :math:`m \times p` array of supervision.
-        :return: Metric computed as a zero-dimensional array
+            compare against ``reference_data``, containing an :math:`m \times d` array
+            of data
+        :return: Computed metric as a zero-dimensional array
         """
 
 
-class MMD(Metric, Generic[_Data]):
+class MMD(Metric[_Data]):
     r"""
-    Definition and calculation of the maximum mean discrepancy metric.
+    Definition and calculation of the (weighted) maximum mean discrepancy metric.
 
-    For a dataset of ``n`` points in ``d`` dimensions, :math:`\mathcal_{D}_1`, and
-    another dataset :math:`\mathcal_{D}_2` of ``m`` points in ``d`` dimensions, the
-    maximum mean discrepancy is given by:
+    For a dataset :math:`\mathcal{D}_1` of ``n`` points in ``d`` dimensions, and
+    another dataset :math:`\mathcal{D}_2` of ``m`` points in ``d`` dimensions, the
+    (weighted) maximum mean discrepancy is given by:
 
-    .. math::
+    .. math:
```

Comment:
Need a double colon at end.

Reply:
I've made this mistake in a few other areas as well (not directly related to the pull), will include those fixes in this pull. Its unfortunate that the docs build without warning, just completely skipping the directives.